core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java

```diff
@@ -173,7 +177,9 @@ private TabletsMetadata buildNonRoot(AccumuloClient client) {
             new IsolatedScanner(client.createScanner(resolvedTable, Authorizations.EMPTY));
         scanner.setRange(range);
 
-        if (checkConsistency && !fetchedCols.contains(ColumnType.PREV_ROW)) {
+        boolean extentsAreNull = extentsToFetch == null;
+
+        if ((checkConsistency && !fetchedCols.contains(ColumnType.PREV_ROW)) || !extentsAreNull) {
```

Comment:
for some reason this causes tests to hang. at least `MiniAccumuloClusterStartStopTest.multipleStopsIsAllowed()`

Reply:
I can think of two possible causes.  One is that there is a bug in these changes.  I have been looking for bugs in the changes and have not seen anything so far.  Another possible cause is there there is code that relies on the current behavior and its only setting the end row for the tablet it wants to retrieve.  I have not looked for this 2nd case yet, would need to look at all usages of these Ample methods.

Reply:
I will help investigate.

Reply:
Some server logs that might help. Multiple assignment failures all with the same exception of DIR not being fetched. The only thing I do to produce these was start up an accumulo instance with uno.  I did add some extra logging in TabletsMetadata so some line numbers might not match up.


<details>

```
Failed to verify tablet !0;~<

java.lang.IllegalStateException: DIR was not fetched
	at com.google.common.base.Preconditions.checkState(Preconditions.java:591)
	at org.apache.accumulo.core.metadata.schema.TabletMetadata.ensureFetched(TabletMetadata.java:158)
	at org.apache.accumulo.core.metadata.schema.TabletMetadata.getDirName(TabletMetadata.java:239)
	at org.apache.accumulo.tserver.TabletServer.checkTabletMetadata(TabletServer.java:939)
	at org.apache.accumulo.tserver.AssignmentHandler.run(AssignmentHandler.java:106)
	at org.apache.accumulo.tserver.ActiveAssignmentRunnable.run(ActiveAssignmentRunnable.java:63)
	at org.apache.htrace.wrappers.TraceRunnable.run(TraceRunnable.java:57)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

```
Caught an Exception in Thread[metadata tablet assignment-Worker-3,5,main]. Thread is dead.
java.lang.RuntimeException: java.lang.IllegalStateException: DIR was not fetched
	at org.apache.accumulo.tserver.AssignmentHandler.run(AssignmentHandler.java:136)
	at org.apache.accumulo.tserver.ActiveAssignmentRunnable.run(ActiveAssignmentRunnable.java:63)
	at org.apache.htrace.wrappers.TraceRunnable.run(TraceRunnable.java:57)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.IllegalStateException: DIR was not fetched
	at com.google.common.base.Preconditions.checkState(Preconditions.java:591)
	at org.apache.accumulo.core.metadata.schema.TabletMetadata.ensureFetched(TabletMetadata.java:158)
	at org.apache.accumulo.core.metadata.schema.TabletMetadata.getDirName(TabletMetadata.java:239)
	at org.apache.accumulo.tserver.TabletServer.checkTabletMetadata(TabletServer.java:939)
	at org.apache.accumulo.tserver.AssignmentHandler.run(AssignmentHandler.java:106)
	... 5 more
```
</details>

Reply:
Thanks @Manno15,  that exception was very helpful, I see what the problem is.

When a tablet is being loaded, the following line fetches the tablet metadata.

https://github.com/apache/accumulo/blob/f5df4f9681736d35e14b92c58033a1cf062b3d4e/server/tserver/src/main/java/org/apache/accumulo/tserver/AssignmentHandler.java#L104

The call to readTablet above does not specify any columns, therefore this means to fetch all columns.  When we make the call to fetch the prev row column, it ends up limiting the fetch to just that column in this case.  I'll make a comment elsewhere with what I think we need to do.

Reply:
Not completely sure the following is correct, we need to something like that.

```suggestion
        if (!fetchedCols.isEmpty() &&(checkConsistency || !extentsAreNull)) {
```

Reply:
Yea this seems right. Tests pass with this change.